airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/spreadsheet.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from typing import List, Mapping
+
+from pygsheets import Spreadsheet, Worksheet
+from pygsheets.client import Client as pygsheets_client
+from pygsheets.exceptions import WorksheetNotFound
+
+
+class GoogleSheets:
+    def __init__(self, client: pygsheets_client, spreadsheet_id: str):
+        self.client = client
+        self.spreadsheet_id = spreadsheet_id
+
+    @property
+    def spreadsheet(self) -> Spreadsheet:
+        """
+        Returns pygsheets.Spreadsheet with opened target spreadsheet by key.
+        """
+        return self.client.open_by_key(self.spreadsheet_id)
+
+    def open_worksheet(self, stream_name: str) -> Worksheet:
+        """
+        Opens the connection to target worksheet, if exists. Otherwise, creates one.
+        """
+        try:
+            stream = self.spreadsheet.worksheet_by_title(stream_name)
+        except WorksheetNotFound:
+            stream = self.spreadsheet.add_worksheet(stream_name)
+        finally:
+            return stream
+
+    def clean_worksheet(self, stream_name: str):
+        """
+        Cleans up the existing records inside the worksheet or creates one, if doesn't exist.
+        """
+        try:
+            stream = self.open_worksheet(stream_name)
+            stream.clear()
+        except WorksheetNotFound:
+            self.spreadsheet.add_worksheet(stream_name)
+
+    def set_headers(self, stream_name: str, headers_list: List[str]):
+        """
+        Sets headers belonging to the input stream
+        """
+        stream: Worksheet = self.open_worksheet(f"{stream_name}")
+        stream.update_row(1, headers_list)
+
+    def index_cols(self, stream: Worksheet) -> Mapping[str, int]:
+        """
+        Helps to find the index of every colums exists in worksheet.
+        Returns: Mapping with column name and it's index.
+            {"id": 1, "name": 2, ..., "other": 99}
+        """
+        header = stream[1]
+        col_index = {}
+        for i, col in enumerate(header):
+            col_index[col] = i + 1
+        return col_index
+
+    def find_duplicates(self, stream: Worksheet, primary_key: str):
```

Comment:
I am actually not sure if it applicable for spreadsheets at all.

It is expected that spreadsheets can have similar rows.

Reply:
It's applicable and works as expected.